packages/@lwc/engine-core/src/framework/check-version-mismatch.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isNull, isUndefined, LWC_VERSION } from '@lwc/shared';
+import { Template } from './template';
+import { getComponentTag } from '../shared/format';
+import { VM } from './vm';
+import { StylesheetFactory } from './stylesheet';
+import { LightningElementConstructor } from './base-lightning-element';
+
+let warned = false;
+
+if (process.env.NODE_ENV === 'development') {
+    // @ts-ignore
+    window.__lwcResetWarnedOnVersionMismatch = () => {
+        warned = false;
+    };
+}
+
+/**
+ * Validate a template, stylesheet, or component to make sure that its compiled version matches
+ * the version used by the LWC engine at runtime.
+ */
+export function checkVersionMismatch(
+    func: Template | StylesheetFactory | LightningElementConstructor,
+    vm?: VM
+) {
+    const versionMatcher = func.toString().match(/\/\*LWC compiler v([\d.]+)\*\//);
```

Comment:
The comment string generation is duplicated in the JS, HTML and the CSS transforms. The regex is defined here in the engine. What about moving both the comment string and regex to `@lwc/shared`, to centralize everything and any issue where the comment string generation gets out of sync from the regular expression?